pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BasePartitionUpsertMetadataManager.java

```diff
@@ -86,6 +87,9 @@ public abstract class BasePartitionUpsertMetadataManager implements PartitionUps
   // If upsertTTL enabled, we will keep track of largestSeenComparisonValue to compute expired segments.
   protected volatile double _largestSeenComparisonValue;
 
+  // Used to initialize a reference to previous row for merging in partial upsert
+  protected LazyRow _reusePreviousRow;
```

Comment:
This can be moved to `ConcurrentMapPartitionUpsertMetadataManager`, and can be made `final`